cmake/Modules/FindAzureSDK_EP.cmake

```diff
@@ -83,38 +85,86 @@ if (NOT AZURESDK_FOUND)
     endif()
 
     if (WIN32)
-      set(CFLAGS_DEF "${CMAKE_C_FLAGS}")
-      set(CXXFLAGS_DEF "${CMAKE_CXX_FLAGS}")
+      set(CFLAGS_DEF " /EHsc -I${TILEDB_EP_INSTALL_PREFIX}/include /Dazure_storage_lite_EXPORTS /DCURL_STATICLIB=1 /DWIN32 ${CMAKE_C_FLAGS}")
+      set(CXXFLAGS_DEF " /EHsc -I${TILEDB_EP_INSTALL_PREFIX}/include /Dazure_storage_lite_EXPORTS /DCURL_STATICLIB=1 /DWIN32 ${CMAKE_CXX_FLAGS}")
+      set(CURL_INCLUDE_DIR "${TILEDB_EP_INSTALL_PREFIX}/include")
     else()
-      set(CFLAGS_DEF "${CMAKE_C_FLAGS} -fPIC")
-      set(CXXFLAGS_DEF "${CMAKE_CXX_FLAGS} -fPIC")
+      #put our switch first in case other items are empty, leaving problematic blank space at beginning
+      set(CFLAGS_DEF "-fPIC ${CMAKE_C_FLAGS}")
+      set(CXXFLAGS_DEF "-fPIC ${CMAKE_CXX_FLAGS}")
     endif()
-
-    ExternalProject_Add(ep_azuresdk
-      PREFIX "externals"
-      # Set download name to avoid collisions with only the version number in the filename
-      DOWNLOAD_NAME ep_azuresdk.zip
-      URL "https://github.com/Azure/azure-storage-cpplite/archive/v0.2.0.zip"
-      URL_HASH SHA1=058975ccac9b60b522c9f7fd044a3d2aaec9f893
-      CMAKE_ARGS
-        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-        -DBUILD_SHARED_LIBS=OFF
-        -DBUILD_TESTS=OFF
-        -DBUILD_SAMPLES=OFF
-        -DCMAKE_PREFIX_PATH=${TILEDB_EP_INSTALL_PREFIX}
-        -DCMAKE_INSTALL_PREFIX=${TILEDB_EP_INSTALL_PREFIX}
-        -DCMAKE_CXX_FLAGS=-fPIC
-        -DCMAKE_C_FLAGS=-fPIC
-      PATCH_COMMAND
-        patch -N -p1 < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_azuresdk/remove-uuid-dep.patch &&
-        patch -N -p1 < ${TILEDB_CMAKE_INPUTS_DIR}/patches/ep_azuresdk/azurite-support.patch
-      LOG_DOWNLOAD TRUE
-      LOG_CONFIGURE TRUE
-      LOG_BUILD TRUE
-      LOG_INSTALL TRUE
-      LOG_OUTPUT_ON_FAILURE ${TILEDB_LOG_OUTPUT_ON_FAILURE}
-      DEPENDS ${DEPENDS}
+    
+    if (WIN32)
+        # needed for applying patches on windows
+        find_package(Git REQUIRED)
+        #see comment on this answer - https://stackoverflow.com/a/45698220
+        #and this - https://stackoverflow.com/a/62967602 (same thread, different answer/comment)
+    else()
+        find_package(Git REQUIRED)
```

Comment:
I would prefer to minimize dependencies, so let's not require `Git` for non-WIN32 and continue to use the unix `patch` command.

Reply:
I was assuming that git already had to be present for the build tree to have been fetched, and hence no new dependency, and thought it might be better to use the same tool to apply a patch rather than trying to have a patch that is acceptable to two different tools, since the two tools apparently don't always accept each other's input.
But that said, can change back to 'patch' if you disagree.

Reply:
Just a quick note, there are many places where we do have the build tree from git itself. We often download github archives of a tag for building. This is how it works for conda builds and any place we use cmake external project ([TileDB-VCF](https://github.com/TileDB-Inc/TileDB-VCF/blob/aba4e5558c8287b349df8330afe3cedff36a6b4d/libtiledbvcf/cmake/Modules/FindTileDB_EP.cmake#L80), [TileDB-MariaDB](https://github.com/TileDB-Inc/TileDB-MariaDB/blob/da4649c68e837386f5aad15eec9834e9f4dfc57e/CMakeLists.txt#L60), etc.)

Reply:
Fine by me. Resolving.